cdap-integration-test/src/main/java/co/cask/cdap/test/remote/RemoteStreamManager.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *  
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.test.remote;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.flow.flowlet.StreamEvent;
+import co.cask.cdap.client.StreamClient;
+import co.cask.cdap.client.config.ClientConfig;
+import co.cask.cdap.common.exception.StreamNotFoundException;
+import co.cask.cdap.test.StreamManager;
+import com.google.common.base.Charsets;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation of {@link StreamManager} that interacts with stream service via HTTP calls
+ */
+public class RemoteStreamManager implements StreamManager {
```

Comment:
This entire class is just a copy of the existing ``RemoteStreamWriter``, with the addition of the ``getEvents`` method. The ``RemoteStreamWriter`` now delegates to this class